### PR TITLE
[10.1.x] TestModuleRepository relies on package-local stuff located in another…

### PR DESCRIPTION
**Pull request by anistor**
… java module

* refactor to avoid module access violations
* make TestModuleRepository final
* generify several usages of ComponentAccessor

### CODE REVIEW

File: core/src/main/java/org/infinispan/factories/ComponentRegistry.java [hunk 1]
```diff
@@ -166,7 +166,7 @@ protected Log getLog() {
          return null;
 
       Class<?> componentClass = componentRef.wired().getClass();
-      ComponentAccessor metadata = moduleRepository.getComponentAccessor(componentClass.getName());
+      ComponentAccessor<Object> metadata = moduleRepository.getComponentAccessor(componentClass.getName());
```
[the remark this answers is not in the document]
Comment (anistor) [reply core/src/main/java/org/infinispan/factories/ComponentRegistry.java:169]: why? that method already returns ComponentAccessor<Object>

**Comment by anistor** [comment]
> `ModuleRepository`

I suppose whoever does that will have their own instance of ModuleRepository that they can do whatever with it.  Will that have any security impact on any running cache manager ?

**Comment by danberindei** [comment]
@anistor yes, there is an impact. Any `ComponentAccessor<T>` instance can invoke the injection/lifecycle methods on any instance of `T`, regardless of whether that instance is registered in any (global) component registry.

File: core/src/main/java/org/infinispan/factories/ComponentRegistry.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (danberindei) [reply core/src/main/java/org/infinispan/factories/ComponentRegistry.java:169]: Hmmm, you're right. I guess making it `ComponentAccessor<Object>` helped centralize the casting in one place, even though it's not technically correct.

**Comment by tristantarrant** [comment]
Not backporting this.

**Pull request closed without merging**